### PR TITLE
Coverity Scan fixes

### PR DESCRIPTION
**Pull request by ondrejholy**
Proposing various fixes based on Coverity Scan results, see commit descriptions for more details.

### CODE REVIEW

File: libfreerdp/crypto/per.c [hunk 1]
```diff
@@ -328,8 +329,8 @@ BOOL per_read_object_identifier(wStream* s, BYTE oid[6])
 		return FALSE;
 
 	Stream_Read_UINT8(s, t12); /* first two tuples */
-	a_oid[0] = (t12 >> 4);
```
Comment (akallabeth) [inline libfreerdp/crypto/per.c]: Where do you get the change from the shift by 4 (division by 16) to division by 40?

File: libfreerdp/crypto/per.c [hunk 2]
```diff
@@ -352,11 +353,12 @@ BOOL per_read_object_identifier(wStream* s, BYTE oid[6])
  * Write PER OBJECT_IDENTIFIER (OID)
  * @param s stream
  * @param oid object identifier (oid)
+ * @warning It works correctly only for limited set of OIDs.
  */
 
 void per_write_object_identifier(wStream* s, BYTE oid[6])
 {
-	BYTE t12 = (oid[0] << 4) & (oid[1] & 0x0F);
+	BYTE t12 = oid[0] * 40 + oid[1];
```
Comment (akallabeth) [inline libfreerdp/crypto/per.c]: Where do you get the change from the shift by 4 (division by 16) to division by 40?

File: libfreerdp/core/security.c [hunk 3]
```diff
@@ -524,9 +524,9 @@ static void fips_expand_key_bits(BYTE* in, BYTE* out)
 		p = b / 8;
 		r = b % 8;
 
-		if (r == 0)
+		if (r <= 1)
 		{
-			out[i] = buf[p] & 0xfe;
+			out[i] = (buf[p] << r) & 0xfe;
```
Comment (akallabeth) [inline libfreerdp/core/security.c]: Why did you add the shift here?

File: libfreerdp/crypto/per.c (hunk 1 above)
Comment (mfleisz) [reply libfreerdp/crypto/per.c]: Afaict this is described in https://msdn.microsoft.com/en-us/library/windows/desktop/bb540809 - the previous calculation with shifting seems to be wrong

File: libfreerdp/core/security.c (hunk 3 above)
Comment (mfleisz) [reply libfreerdp/core/security.c]: Good question! :)

File: libfreerdp/crypto/per.c (hunk 1 above)
Comment (akallabeth) [reply libfreerdp/crypto/per.c]: Ok, so that explains it, perfect.

File: libfreerdp/core/security.c (hunk 3 above)
Comment (akallabeth) [reply libfreerdp/core/security.c]: Ok, missed the commit message in overview mode, so this looks good too.

Comment (ondrejholy) [reply libfreerdp/core/security.c]: In order to handle the case when r=1 (without access into buf[p+1])...

File: libfreerdp/crypto/per.c (hunk 1 above)
Comment (ondrejholy) [reply libfreerdp/crypto/per.c]: First tuple is encoded as 40m + n if m < 3 and n < 40, see:
https://msdn.microsoft.com/en-us/library/windows/desktop/bb540809
https://msdn.microsoft.com/en-us/library/cc240836.aspx

Sorry that I did not explain it fully in the commit message.

File: libfreerdp/core/security.c (hunk 3 above)
Comment (mfleisz) [reply libfreerdp/core/security.c]: Ah - alright - understood!

File: libfreerdp/core/rdp.c [hunk 4]
```diff
@@ -615,17 +615,11 @@ BOOL rdp_send_data_pdu(rdpRdp* rdp, wStream* s, BYTE type, UINT16 channel_id)
...
-	size_t sec_hold;
 	UINT32 pad;
 
 	length = Stream_GetPosition(s);
 	Stream_SetPosition(s, 0);
 	rdp_write_header(rdp, s, length, rdp->mcs->messageChannelId);
-	sec_bytes = rdp_get_sec_bytes(rdp, sec_flags);
-	sec_hold = Stream_GetPosition(s);
-	Stream_Seek(s, sec_bytes);
-	Stream_SetPosition(s, sec_hold);
```
Comment (bmiklautz) [inline libfreerdp/core/rdp.c]: Nice finding. Looks like a rather old copy 'n' paste error ;)

**Comment by bmiklautz** [comment]
@ondrejholy nice one! Thank you.
